zerver/views/message_fetch.py

```diff
@@ -78,6 +78,15 @@ def get_search_fields(
     }
 
 
+def clean_narrow_for_web_public_api(narrow: OptionalNarrowListT) -> OptionalNarrowListT:
+    if narrow is None:
+        return None
+
+    # Remove {'operator': 'in', 'operand': 'home', 'negated': False} from narrow.
+    # This is to allow spectators to access all messages. The narrow should still pass
+    # is_web_public_narrow check after this change.
+    return [term for term in narrow if (term["operator"] != "in" and term["operand"] != "home" and not term["negated"])]
```

Comment:
This might be better written with an `if not (`  rather than all the individual terms being negated.

Reply:
Sorrry, updated!